File: ZezeJava/ZezeJava/src/main/java/Zeze/World/WorldStatic.java
package Zeze.World;

import Zeze.Arch.ProviderUserSession;

public class WorldStatic extends AbstractWorldStatic {
    public final World world;

    public WorldStatic(World world) {
        this.world = world;
        var providerApp = world.providerApp;

        RegisterProtocols(providerApp.providerService);
        RegisterZezeTables(providerApp.zeze);
    }

    @Override
    protected long ProcessSwitchWorldRequest(Zeze.Builtin.World.Static.SwitchWorld r) throws Exception {
        var session = ProviderUserSession.get(r);

        var instanceId = world.getMapManager().enterMap(session, r.Argument);
        r.Result.setMapInstanceId(instanceId);

        session.sendResponseDirect(r);
        return 0;
    }
}

File: Components/WorldDemo/Zeze/World/WorldStatic.cs

using Zeze.Net;

namespace Zeze.World
{
    /// <summary>
    /// 这个模块用来定义地图组件需要的静态绑定的协议。
    /// 现在只有服务器处理的SwitchWorld协议，由于发送代码写在哪里跟定义在那个模块无关，
    /// 这个协议的发送代码写到World模块中了。
    /// 以后，当需要增加客户端处理的协议时，处理入口会在这个模块里面。
    /// </summary>
    public class WorldStatic : AbstractWorldStatic
    {
        public WorldStatic(Service service)
        {
            RegisterProtocols(service);
        }
    }
}
